cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20AccessTokenController.java

```diff
@@ -37,137 +28,72 @@
 public final class OAuth20AccessTokenController extends BaseOAuthWrapperController {
 
     @Autowired
-    @Qualifier("accessTokenJwtGenerator")
-    private JwtGenerator accessTokenJwtGenerator;
+    @Qualifier("defaultAccessTokenFactory")
+    private AccessTokenFactory accessTokenFactory;
 
-    /**
-     * Instantiates a new o auth20 access token controller.
-     */
-    public OAuth20AccessTokenController() {}
-
-    /**
-     * Ensure the encryption secret has been set.
-     */
-    @PostConstruct
-    public void postConstruct() {
-        CommonHelper.assertNotNull("encryptionSecret", accessTokenJwtGenerator.getEncryptionSecret());
-    }
+    @NotNull
+    @Value("${tgt.timeToKillInSeconds:7200}")
```

Comment:
You raise a good point here. For simplicity, it's currently the same timeout as for the TGT, but it should be optionnally changed. I just opened: https://github.com/Jasig/cas/issues/1505 to track that improvement (we can also have the same question for OAuth codes). It will be revisited at least when dealing with refresh tokens (another timeout).